addons/website_forum/models/forum_post.py

```diff
@@ -868,3 +868,37 @@ def _search_render_results(self, fetch_fields, mapping, icon, limit):
             if with_date:
                 data['date'] = self.env['ir.qweb.field.date'].record_to_html(post, 'write_date', {})
         return results_data
+
+    def _get_related_posts(self, limit=5):
+        """Return at most a list of 5 posts related to the main post, based on tag
+        Jaccard similarity. It computes similarity of sets based on ratio of sets
+        intersection divided by sets union (and thus varies from 0 to 1, 1 being
+        identical sets)."""
+
+        if not self.tag_ids:
+            return self.env['forum.forum']
+
+        self.ensure_one()
+        self.env.cr.execute(SQL("""
+            SELECT forum_post.id,
+              -- Jaccard similarity
+                   (COUNT(DISTINCT intersection_tag_rel.forum_tag_id))::DECIMAL
+                   / COUNT(DISTINCT union_tag_rel.forum_tag_id)::DECIMAL AS similarity
```

Comment:
So it's basically comparing 2 posts with "how many tags we have in common" divided by "how many different tags we have in total" ?